### PR TITLE
`azurerm_cognitive_deployment` - remove limitation on capacity and update document

### PR DESCRIPTION
**Pull request by liuwuliuyun**
`azurerm_cognitive_deployment` 

- remove limitation on capacity and update document
- fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/22135

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
LGTM 💾

File: internal/services/cognitive/cognitive_deployment_resource.go [hunk 1]
```diff
@@ -146,11 +146,10 @@ func (r CognitiveDeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 						ForceNew: true,
 					},
 					"capacity": {
-						Type:         pluginsdk.TypeInt,
-						Optional:     true,
-						ForceNew:     true,
-						Default:      1,
-						ValidateFunc: validation.IntBetween(1, 10000),
```
Comment (katbyte) [inline internal/services/cognitive/cognitive_deployment_resource.go:153]: actually shouldn't this be validating that it is greater then 0 and not allow negative values?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/cognitive/cognitive_deployment_resource.go (hunk 1 above)
Comment (liuwuliuyun) [reply internal/services/cognitive/cognitive_deployment_resource.go:153]: Sure, let me update this one

**Review by manicminer — APPROVED** [review]
Thanks @liuwuliuyun, LGTM

**Pull request merged**